packages/flutter/lib/src/animation/tween_sequence.dart

```diff
@@ -0,0 +1,128 @@
+import 'package:flutter/foundation.dart';
+
+import 'animation.dart';
+import 'curves.dart';
+import 'tween.dart';
+
+/// Enables creating an [Animation] whose value is defined by a
+/// sequence of [Tween]s.
+///
+/// Each [TweenSequenceItem] has a weight that defines its percentage
+/// of the animation's duration, and an optional curve that's applied
+/// to the item's tween. Each tween defines the animation's value
+/// during the interval indicated by its weight.
+///
+/// For example, to define an animation that uses an easing curve to
+/// interpolate between 5.0 and 10.0 during the first 40% of the
+/// animation, remain at 10.0 for the next 20%, and then return to
+/// 10.0 for the final 40%:
+///
+/// ```
+/// final Animation<double> = new TweenSequence(
+///   <TweenSequenceItem<double>>[
+///     new TweenSequenceItem<double>(
+///       tween: new Tween<double>(begin: 5.0, end: 10.0),
+///       curve: Curves.ease
+///       weight: 40.0,
+///     ),
+///     new TweenSequenceItem<double>(
+///       tween: new ConstantTween<double>(10.0),
+///       weight: 20.0,
+///     ),
+///     new TweenSequenceItem<double>(
+///       tween: new Tween<double>(begin: 10.0, end: 5.0),
+///       curve: Curves.ease,
+///       weight: 40.0,
+///     ),
+///   ],
+/// ).animate(myAnimationController);
+///```
+class TweenSequence<T> extends Animatable<T> {
+  /// Construct a TweenSequence.
+  ///
+  /// The [items] parameter must be a list of one or more
+  /// [TweenSequenceItem]s.
+  TweenSequence(List<TweenSequenceItem<T>> items) : assert(items != null && items.isNotEmpty) {
+    _items.addAll(items);
+
+    double totalWeight = 0.0;
+    for (TweenSequenceItem<T> item in _items)
+      totalWeight += item.weight;
+
+    double start = 0.0;
+    for (int i = 0; i < _items.length; i++) {
+      final double end = start + _items[i].weight / totalWeight;
+      _intervals.add(new _Interval(start, end));
+      start = end;
```

Comment:
I'm a little concerned about rounding errors here. they'll add up as you add each item, and you won't reach precisely 1.0. Maybe keep track of the sum of all the weights so far and use that to compute the end, rather than adding start? Though that still won't guarantee reaching 1.0...

Reply:
Good point. I've forced the end of the last item to be 1.0.